front/app/hooks/useNavbarItemEnabled.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react';
+import { navbarItemsStream, TDefaultNavbarItemCode } from 'services/navbar';
+import { isNilOrError } from 'utils/helperUtils';
+
+export default function useNavbarItemEnabled(code: TDefaultNavbarItemCode) {
+  const [itemEnabled, setItemEnabled] = useState<
+    undefined | null | Error | boolean
+  >(undefined);
+
+  useEffect(() => {
+    const subscription = navbarItemsStream().observable.subscribe(
+      (response) => {
+        if (isNilOrError(response)) {
+          setItemEnabled(response);
+          return;
+        }
+
+        setItemEnabled(
+          response.data.some(
+            (navbarItem) => code === navbarItem.attributes.code
+          )
+        );
```

Comment:
I am a bit confused about this- it is an API response, so it can always be an error. Odd that `streams.ts` doesn't automatically add that as a possible type. The same problem seems to exist in many other hooks- see e.g. `useProject` or `useIdea`, where we also check for `isNilOrError` even though the response doesn't have `Error` as a type. Should I add it to issues / tech debt?